iodata/formats/qchemlog.py

```diff
@@ -0,0 +1,368 @@
+# IODATA is an input and output module for quantum chemistry.
+# Copyright (C) 2011-2019 The IODATA Development Team
+#
+# This file is part of IODATA.
+#
+# IODATA is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 3
+# of the License, or (at your option) any later version.
+#
+# IODATA is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>
+# --
+"""Q-Chem Log file format.
+
+This module will load Q-Chem log file into IODATA.
+"""
+
+
+import re
+from typing import Tuple
+
+import numpy as np
+
+from ..docstrings import document_load_one
+from ..orbitals import MolecularOrbitals
+from ..periodic import sym2num
+from ..utils import LineIterator, kcalmol, calmol
+
+__all__ = []
+
+PATTERNS = ['*.qchemlog']
+
+
+@document_load_one("qchemlog", ['atcoords', 'atmasses', 'atnums', 'charge', 'energy', 'g_rot',
+                                'mo', 'lot', 'nelec', 'obasis_name', 'run_type', 'extra'],
+                   ['athessian'])
+def load_one(lit: LineIterator) -> dict:
+    """Do not edit this docstring. It will be overwritten."""
+    data = load_qchemlog_low(lit)
+
+    # add these labels if they are loaded
+    result_labels = ['atcoords', 'atmasses', 'atnums', 'charge', 'charge', 'energy', 'g_rot',
+                     'run_type', 'athessian', 'lot', 'obasis_name']
+    result = {label: data[label] for label in result_labels if data.get(label) is not None}
+    # add number of electrons
+    result['nelec'] = data['alpha_elec'] + data['beta_elec']
+
+    # mulliken charges
+    if data.get("mulliken_charges") is not None:
+        result["atcharges"] = {"mulliken": data["mulliken_charges"]}
+
+    # build molecular orbitals
+    # ------------------------
+    # restricted case
+    if not data['unrestricted']:
+        mo_energies = np.concatenate((data['mo_a_occ'], data['mo_a_vir']), axis=0)
+        mo_coeffs = np.empty((data['nbasis'], data['norba'])) * np.nan
```

Comment:
Would there be a way to avoid creating a large matrix with nans? E.g. would the following be ok?
```suggestion
        mo_coeffs = np.empty((0, data['norba']))
```
If not, it would be good for readability to use [np.full](https://numpy.org/doc/stable/reference/generated/numpy.full.html) instead of multiplying with nan.

Reply:
Thanks for noticing this, Toon! I think it would be great to keep the right shape, instead of return an empty array with shape `(0, data['norba'])`. We can try 

```python
np.full((data['nbasis'], data['norba']), np.nan)
```

as you suggested, which is cleaner and faster. 
Would this be good? @tovrstra

Reply:
OK, Good for me.

We should address this in future though, e.g. by allowing `mo_coeffs=None` when creating a `MolecularOrbitals` instance. This would be more in line with the data (not) present in the qhem log file and it would be efficient in terms of memory. I've opened an issue for it, which we can discuss later. (#168)  This way, we can merge your PR more quickly.

Reply:
This make more sense. I have fixed the codes with `np.full()` which is more efficient, 45eeba0.

We can discuss more in GitHub issues and thanks for giving this nice point!